src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -964,50 +956,33 @@ internal static void CreateNameToMembersMap
         {
             var dictionary = new Dictionary<string, ImmutableArray<TNamedTypeSymbol>>(comparer);
 
-            foreach (var kvp in map)
-            {
-                ImmutableArray<TNamespaceOrTypeSymbol> members = kvp.Value;
+            foreach (var (name, members) in map)
+                dictionary.Add(name, getOrCreateNamedTypes(members));
 
-                bool hasType = false;
-                bool hasNamespace = false;
```

Comment:
no need for any tracking flags, or having to deal with any of the combinations of false/true, true/false, true/true (false/false can't happen as the value is always present and has at least one element it represents)